python/vyos/ifconfig/veth.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2022 VyOS maintainers and contributors <maintainers@vyos.io>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+from vyos.ifconfig.interface import Interface
+
+@Interface.register
+class VethIf(Interface):
+    """
+    Abstraction of a Linux veth interface
+    """
+    iftype = 'veth'
+    definition = {
+        **Interface.definition,
+        **{
+            'section': 'virtual-ethernet',
+            'prefixes': ['veth', ],
+            'bridgeable': True,
+        },
+    }
+
+    def _create(self):
+        conf = self.config
+        peer_name = conf["peer"]["name"]
+        cmd = f'ip link add {conf["ifname"]} type {conf["type"]} peer name {peer_name}'
```

Comment:
Please keep the common python code we have in other interfaces

```python
        """
        Create veth interface in OS kernel. Interface is administrative
        down by default.
        """
        # create virtual-ethernet interface
        cmd  = 'ip link add {ifname} type {type}'.format(**self.config)' 
        cmd += f' peer name {self.config["peer"]["name"]}'
        self._cmd(cmd)

        # interface is always A/D down. It needs to be enabled explicitly
        self.set_admin_state('down')
```

Reply:
Thanks, done!